File: lib/domain/auth/gateways/auth_test_login_gateway.rb
# frozen_string_literal: true

module Domain
  module Auth
    module Gateways
      module AuthTestLoginGateway
        # @param input_dto [Domain::Auth::Dtos::AuthTestMockLoginInputDto]
        # @return [Domain::Auth::Dtos::AuthTestMockLoginPersistResult]
        def persist_mock_user_and_session!(input_dto)
          raise NotImplementedError, "#{self.class}##{__method__}"
        end
      end
    end
  end
end

File: crates/agrr-domain/src/auth/gateways/auth_test_login_gateway.rs
use crate::auth::dtos::{AuthTestMockLoginInput, AuthTestMockLoginPersistResult};

/// Ruby: `Domain::Auth::Gateways::AuthTestLoginGateway`
pub trait AuthTestLoginGateway: Send + Sync {
    fn persist_mock_user_and_session(
        &self,
        input_dto: &AuthTestMockLoginInput,
    ) -> AuthTestMockLoginPersistResult;
}
